shap/plots/_beeswarm.py

```diff
@@ -188,7 +189,7 @@ def beeswarm(
     if ax is None:
         ax = pl.gca()
     fig = ax.get_figure()
-    assert fig is not None  # type narrowing for mypy
+    assert isinstance(fig, Figure)  # type narrowing for mypy
```

Comment:
It seems like in matplotlib 3.10 `ax.get_figure()` can return a subfigure which does not have a "set_size_inches" method, that we'll use in lines 362 cf. Therefore we check here directly if `fig` has the correct type.